nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/controller/flowanalysis/AbstractFlowAnalysisRuleNode.java

```diff
@@ -0,0 +1,369 @@
+        }
+    }
+
+    @Override
+    public void verifyCanEnable(final Set<ControllerServiceNode> ignoredServices) {
+        if (isEnabled()) {
+            throw new IllegalStateException("Cannot enable " + getFlowAnalysisRule().getIdentifier() + " because it is not disabled");
+        }
+
+        final Collection<ValidationResult> validationResults = getValidationErrors(ignoredServices);
+        if (!validationResults.isEmpty()) {
+            throw new IllegalStateException(this + " cannot be enabled because it is not currently valid");
+        }
+    }
+
+    @Override
+    public void verifyCanUpdate() {
+        if (isEnabled()) {
+            throw new IllegalStateException("Cannot update " + getFlowAnalysisRule().getIdentifier() + " because it is currently enabled");
+        }
+    }
+
+    @Override
+    public void verifyCanClearState() {
+        verifyCanUpdate();
+    }
+
+    @Override
+    public String getProcessGroupIdentifier() {
+        return null;
+    }
+
+    @Override
+    public ParameterLookup getParameterLookup() {
+        return ParameterLookup.EMPTY;
+    }
+
+    @Override
+    public String toString() {
+        FlowAnalysisRule flowAnalysisRule = flowAnalysisRuleRef.get().getFlowAnalysisRule();
+        try (final NarCloseable narCloseable = NarCloseable.withComponentNarLoader(getExtensionManager(), flowAnalysisRule.getClass(), flowAnalysisRule.getIdentifier())) {
+            return getFlowAnalysisRule().toString();
+        }
+    }
+
+    @Override
+    public void enable() {
+        verifyCanEnable();
+        setState(FlowAnalysisRuleState.ENABLED, OnEnabled.class);
+    }
+
+    @Override
+    public void disable() {
+        verifyCanDisable();
+        setState(FlowAnalysisRuleState.DISABLED, OnDisabled.class);
+
+        ruleViolationsManager.removeRuleViolationsForRule(getIdentifier());
+        ruleViolationsManager.cleanUp();
+    }
+
+    private void setState(FlowAnalysisRuleState newState, Class<? extends Annotation> annotation) {
+        final ConfigurationContext configContext = new StandardConfigurationContext(this, this.serviceLookup, null, getVariableRegistry());
+
+        try (final NarCloseable nc = NarCloseable.withComponentNarLoader(getExtensionManager(), getFlowAnalysisRule().getClass(), getIdentifier())) {
+            ReflectionUtils.invokeMethodsWithAnnotation(annotation, getFlowAnalysisRule(), configContext);
+
+            this.state = newState;
+
+            LOG.debug("Successfully {} {}", newState.toString().toLowerCase(), this);
+        } catch (Exception e) {
+            final Throwable cause = e instanceof InvocationTargetException ? e.getCause() : e;
+
+            final ComponentLog componentLog = new SimpleProcessLogger(getIdentifier(), getFlowAnalysisRule());
+
+            componentLog.error("Failed to invoke {} method due to {}", annotation.getSimpleName(), cause);
```

Comment:
I followed the pattern in `StandardControllerServiceNode` here as well. I checked the UI to make sure the "due to" part is not lost and the message is reported with the exception on the bulletin as well. It is the case so the "due to" part can be omitted.
I'd not change the pattern otherwise though. The expression `"Failed to invoke method [{}]"` would be confusing as the placeholder doesn't receive the method itself but the annotation of the method.